src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -500,26 +501,25 @@ private async Task ConnectFirstAsync(CancellationToken cancellationToken)
 
                         HConsole.WriteLine(this, $"Failed to connect to address {address}");
 
-                        if (attempt.HasException) // else gather exceptions
+                        if (attempt.HasException)
                         {
                             if (attempt.Exception is RemoteException { Error: RemoteError.HazelcastInstanceNotActive })
                             {
-                                _logger.LogDebug($"Failed to connect to address {address} (member is not active).");
+                                _logger.LogWarning($"Failed to connect to address {address} (member is not active).");
                             }
                             else if (attempt.Exception is TimeoutException)
                             {
-                                _logger.LogDebug($"Failed to connect to address {address} (socket timeout).");
+                                _logger.LogWarning($"Failed to connect to address {address} (socket timeout).");
                             }
                             else
                             {
-                                exceptions ??= new List<Exception>();
-                                exceptions.Add(attempt.Exception);
-
-                                _logger.LogDebug(attempt.Exception, $"Failed to connect to address {address}.");
+                                isExceptionThrown = true;
+                                _logger.LogError(attempt.Exception, "Failed to connect to address {address}.", address.ToString());
                             }
                         }
                         else
                         {
+                            //is !attempt.HasException possible here?
```

Comment:
I am not entirely sure but better handle it, yes. And... if it cannot happen *today* it may happen in the future.